commands/create_kubeconfig.go

```diff
@@ -452,43 +406,37 @@ func createKubeconfig(args createKubeconfigArguments) (createKubeconfigResult, e
 		}
 	} else {
 		// create a self-contained kubeconfig
-		kubeconfig := Kubeconfig{
-			APIVersion:     "v1",
-			Kind:           "Config",
-			CurrentContext: args.contextName,
-			Clusters: []KubeconfigNamedCluster{
-				{
-					Name: "giantswarm-" + args.clusterID,
-					Cluster: KubeconfigCluster{
-						Server:                   result.apiEndpoint,
-						CertificateAuthorityData: base64.StdEncoding.EncodeToString([]byte(response.Payload.CertificateAuthorityData)),
-					},
-				},
-			},
-			Contexts: []KubeconfigNamedContext{
-				{
-					Name: args.contextName,
-					Context: KubeconfigContext{
-						Cluster: "giantswarm-" + args.clusterID,
-						User:    "giantswarm-" + args.clusterID + "-user",
-					},
-				},
-			},
-			Users: []KubeconfigUser{
-				{
-					Name: "giantswarm-" + args.clusterID + "-user",
-					User: KubeconfigUserKeyPair{
-						ClientCertificateData: base64.StdEncoding.EncodeToString([]byte(response.Payload.ClientCertificateData)),
-						ClientKeyData:         base64.StdEncoding.EncodeToString([]byte(response.Payload.ClientKeyData)),
-					},
+		var yamlBytes []byte
+		{
+			c := k8srestconfig.Config{
+				Logger: microloggertest.New(),
+
+				Address:   result.apiEndpoint,
+				InCluster: false,
+				TLS: k8srestconfig.TLSClientConfig{
+					CAData:  []byte(response.Payload.CertificateAuthorityData),
+					CrtData: []byte(response.Payload.ClientCertificateData),
+					KeyData: []byte(response.Payload.ClientKeyData),
 				},
-			},
-		}
+			}
+			restConfig, err := k8srestconfig.New(c)
+			if err != nil {
+				return result, microerror.Mask(err)
+			}
 
-		// create YAML
-		yamlBytes, err := yaml.Marshal(&kubeconfig)
-		if err != nil {
-			return result, microerror.Mask(err)
+			kcc := kubeconfig.Config{
+				Logger: microloggertest.New(),
+				K8sClient: fake.NewSimpleClientset(),
+			}
+
+			kc, err := kubeconfig.New(kcc)
+			if err != nil {
+				return result, microerror.Mask(err)
+			}
+			yamlBytes, err = kc.NewKubeConfigForRESTConfig(nil, restConfig, fmt.Sprintf("giantswarm-%s", args.clusterID), "")
```

Comment:
context should nil? or background?

Reply:
We should provide a background context. Ideally this should be at the entrypoint to the command. Later we might need to use the context for calling other libraries.